src/app.js

```diff
@@ -1,115 +1,78 @@
-var SVG_NS = "http://www.w3.org/2000/svg"
-
 export default function (app) {
-  var view = app.view || function () {
-    return ""
-  }
-
-  var model
+  var state //= app.state || {}
+  var view = app.view
   var actions = {}
-  var hooks = {
-    onError: [],
-    onAction: [],
-    onUpdate: [],
-    onRender: []
-  }
-  var subscriptions = []
-
+  var events = {}
   var node
-  var root
   var element
-  var plugins = app.plugins || []
-
-  for (var i = -1; i < plugins.length; i++) {
-    var plugin = i < 0 ? app : plugins[i](app)
-    var obj = plugin.model
-
-    if (obj != null) {
-      model = merge(model, obj)
-    }
-
-    if (obj = plugin.actions) {
-      init(actions, obj)
-    }
 
-    if (obj = plugin.subscriptions) {
-      subscriptions = subscriptions.concat(obj)
-    }
+  for (var i = -1, plugins = app.plugins || []; i < plugins.length; i++) {
+    var plugin = plugins[i] ? plugins[i](app) : app
 
-    if (obj = plugin.hooks) {
-      Object.keys(obj).map(function (key) {
-        hooks[key].push(obj[key])
-      })
+    if (plugin.state != null) {
```

Comment:
with single line `if`s you can remove curly braces. Unclear if how it minifies though 🤔 

```javascript
if (plugin.state != nulll) state = merge(state, plugin.state)
```